diff -up jwhois-4.0/example/jwhois.conf_old jwhois-4.0/example/jwhois.conf
--- jwhois-4.0/example/jwhois.conf_old	2009-01-23 12:57:35.000000000 +0100
+++ jwhois-4.0/example/jwhois.conf	2009-01-23 12:58:41.000000000 +0100
@@ -171,7 +171,7 @@ whois-servers {
 	"\\.fm$" = "www.dot.fm";
 	"\\.fo$" = "whois.ripe.net";
 	"\\.fr$" = "whois.nic.fr";
-	"\\.gi$" = "www.nic.gi";
+	"\\.gi$" = "whois2.afilias-grs.net";
 	"\\.gov$" = "whois.nic.gov";
 	"\\.gg$" = "whois.isles.net";
 	"\\.gm$" = "whois.ripe.net";
