diff -up jwhois-4.0/example/jwhois.conf_old jwhois-4.0/example/jwhois.conf
--- jwhois-4.0/example/jwhois.conf_old	2007-03-26 11:12:08.000000000 +0200
+++ jwhois-4.0/example/jwhois.conf	2008-10-13 14:43:48.000000000 +0200
@@ -593,13 +593,12 @@ cidr6-blocks {
 	"2001:1600::/23" = "whois.ripe.net";
 	"2001:1800::/23" = "whois.arin.net";
 	"2001:1A00::/23" = "whois.ripe.net";
-	"2001:1C00::/23" = "whois.ripe.net";
-	"2001:1E00::/23" = "whois.ripe.net";
+	"2001:1C00::/22" = "whois.ripe.net";
 	"2001:2000::/20" = "whois.ripe.net";
 	"2001:3000::/21" = "whois.ripe.net";
 	"2001:3800::/22" = "whois.ripe.net";
 	"2001:4000::/23" = "whois.ripe.net";
-	"2001:4200::/23" = "whois.arin.net";
+	"2001:4200::/23" = "whois.afrinic.net";
 	"2001:4400::/23" = "whois.apnic.net";
 	"2001:4600::/23" = "whois.ripe.net";
 	"2001:4800::/23" = "whois.arin.net";
@@ -612,21 +611,15 @@ cidr6-blocks {
 
 	"2003:0000::/18" = "whois.ripe.net";
 
-	"2400:0000::/19" = "whois.apnic.net";
-	"2400:2000::/19" = "whois.apnic.net";
-	"2400:4000::/21" = "whois.apnic.net";
-	"2404:0000::/23" = "whois.apnic.net";
-
-	"2600:0000::/22" = "whois.arin.net";
-	"2604:0000::/22" = "whois.arin.net";
-	"2608:0000::/22" = "whois.arin.net";
-	"260C:0000::/22" = "whois.arin.net";
+	"2400:0000::/12" = "whois.apnic.net";
+
+	"2600:0000::/12" = "whois.arin.net";
 	"2610:0000::/23" = "whois.arin.net";
+	"2620:0000::/23" = "whois.arin.net";
 
-	"2800:0000::/23" = "whois.lacnic.net";
+	"2800:0000::/12" = "whois.lacnic.net";
 
-	"2A00:0000::/21" = "whois.ripe.net";
-	"2A01:0000::/16" = "whois.ripe.net";
+	"2C00:0000::/12" = "whois.afrinic.net";
 
 	#
 	# Experimental IPv6 network 6bone (RFC2471)
